From c47d4b45b932a60dcceb28aca5bb1f601f590010 Mon Sep 17 00:00:00 2001
From: Teemu Takaluoma <teemu.takaluoma@arm.com>
Date: Mon, 1 Jul 2019 14:36:21 +0300
Subject: [PATCH] Use KVStore based storage as default. Remove obsolete SOTP
 configs

New defaults in PAL works out-of-box for most applications, removing
the need for PAL user configuration file in most use cases.
---
 .../Configs/pal_config/mbedOS/mbedOS_SST.h    |  52 ---------
 .../pal_config/mbedOS/mbedOS_default.h        | 105 ++++++++++--------
 mbed-client-pal/Configs/pal_config/sotp_fs.h  |  30 -----
 .../PAL-Impl/Services-API/pal_configuration.h |   2 -
 4 files changed, 59 insertions(+), 130 deletions(-)
 delete mode 100644 mbed-client-pal/Configs/pal_config/mbedOS/mbedOS_SST.h
 mode change 100755 => 100644 mbed-client-pal/Configs/pal_config/mbedOS/mbedOS_default.h
 delete mode 100755 mbed-client-pal/Configs/pal_config/sotp_fs.h

diff --git a/mbed-client-pal/Configs/pal_config/mbedOS/mbedOS_SST.h b/mbed-client-pal/Configs/pal_config/mbedOS/mbedOS_SST.h
deleted file mode 100644
index 13ba39a..0000000
--- a/mbed-client-pal/Configs/pal_config/mbedOS/mbedOS_SST.h
+++ /dev/null
@@ -1,52 +0,0 @@
-/*******************************************************************************
-* Copyright 2016, 2017 ARM Ltd.
-*
-* Licensed under the Apache License, Version 2.0 (the "License");
-* you may not use this file except in compliance with the License.
-* You may obtain a copy of the License at
-*
-*     http://www.apache.org/licenses/LICENSE-2.0
-*
-* Unless required by applicable law or agreed to in writing, software
-* distributed under the License is distributed on an "AS IS" BASIS,
-* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-* See the License for the specific language governing permissions and
-* limitations under the License.
-*******************************************************************************/
-
-#ifndef PAL_MBEDOS_SST_CONFIGURATION_H_
-#define PAL_MBEDOS_SST_CONFIGURATION_H_
-
-/*!
-* \brief This file is for more specific definitions (per board or module if needed).
-*        if this file is defined it will be included from pal_configuration.h
-*        if not, the default file will be included - if needed
-*/
-
-//define sst (kvstore) support flag
-#ifndef MBED_CONF_MBED_CLOUD_CLIENT_EXTERNAL_SST_SUPPORT
-    #define MBED_CONF_MBED_CLOUD_CLIENT_EXTERNAL_SST_SUPPORT
-#endif
-
-//mount fs to MBED_CONF_STORAGE_DEFAULT_KV mount point (same mount point as kvstore defines)
-#ifndef PAL_FS_MOUNT_POINT_PRIMARY
-    #define EXPANSION_STR(x) STR(x) //stringification of macro value
-    #define STR(x) #x //stringification of the macro
-    #define PAL_FS_MOUNT_POINT_PRIMARY "/" EXPANSION_STR(MBED_CONF_STORAGE_DEFAULT_KV)
-#endif
-
-//define secondary mount point to the same mount point as primary
-#ifndef PAL_FS_MOUNT_POINT_SECONDARY
-    #define PAL_FS_MOUNT_POINT_SECONDARY PAL_FS_MOUNT_POINT_PRIMARY
-#endif
-
-//issue a warning if PAL_USE_INTERNAL_FLASH!=0 and or define PAL_USE_INTERNAL_FLASH=0
-#if PAL_USE_INTERNAL_FLASH //PAL_USE_INTERNAL_FLASH != 0
-    #warning "Internal flash APIs should be disabled with KVStore"
-#else
-    #define PAL_USE_INTERNAL_FLASH 0
-#endif
-
-#include "mbedOS_default.h"
-
-#endif /* PAL_MBEDOS_SST_CONFIGURATION_H_ */
diff --git a/mbed-client-pal/Configs/pal_config/mbedOS/mbedOS_default.h b/mbed-client-pal/Configs/pal_config/mbedOS/mbedOS_default.h
old mode 100755
new mode 100644
index df9c3e3..865b1be
--- a/mbed-client-pal/Configs/pal_config/mbedOS/mbedOS_default.h
+++ b/mbed-client-pal/Configs/pal_config/mbedOS/mbedOS_default.h
@@ -1,24 +1,33 @@
-/*******************************************************************************
- * Copyright 2016-2018 ARM Ltd.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- *******************************************************************************/
+// ----------------------------------------------------------------------------
+// Copyright 2018-2019 ARM Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------
 
 #ifndef PAL_MBEDOS_CONFIGURATION_H_
+#define PAL_MBEDOS_CONFIGURATION_H_
 
 #include "cmsis_os.h"
 #include "mbed_version.h"
 
+/*!
+* \brief This file is for more specific definitions (per board or module if needed).
+*        if this file is defined it will be included from pal_configuration.h
+*        if not, the default file will be included - if needed
+*/
+
 #if (defined(MBED_DEBUG) && !defined(DEBUG))
     #define DEBUG
 #endif
@@ -27,61 +36,65 @@
     #define PAL_RTOS_WAIT_FOREVER osWaitForever
 #endif
 
+#if (PAL_NET_DNS_SUPPORT == true) && !(defined(PAL_DNS_API_VERSION))
+#define PAL_DNS_API_VERSION 2 //!< asyncronous DNS API
+#endif
+
+//defines SST (KVstore) support flag.
+//This flag should already be defined as this is client library flag.
+#ifndef MBED_CONF_MBED_CLOUD_CLIENT_EXTERNAL_SST_SUPPORT
+    #warning "External SST support has not been enabled."
+	#define MBED_CONF_MBED_CLOUD_CLIENT_EXTERNAL_SST_SUPPORT 1
+#endif
+
 #ifndef PAL_NUMBER_OF_PARTITIONS
     #define PAL_NUMBER_OF_PARTITIONS 1
 #endif
 
-
+//mount fs to MBED_CONF_STORAGE_DEFAULT_KV mount point (same mount point as kvstore defines)
 #ifndef PAL_FS_MOUNT_POINT_PRIMARY
-    #if (MBED_VERSION >= 51000)  // 5.10 or above
-        #define PAL_FS_MOUNT_POINT_PRIMARY    "/default"           //!< User should change this for the his working folder
-    #else
-        #define PAL_FS_MOUNT_POINT_PRIMARY    "/sd"
-    #endif
+    #define EXPANSION_STR(x) STR(x) //stringification of macro value
+    #define STR(x) #x //stringification of the macro
+    #define PAL_FS_MOUNT_POINT_PRIMARY "/" EXPANSION_STR(MBED_CONF_STORAGE_DEFAULT_KV)
 #endif
 
+//define secondary mount point to the same mount point as primary
 #ifndef PAL_FS_MOUNT_POINT_SECONDARY
-    #if (PAL_NUMBER_OF_PARTITIONS == 2)
-        #if (MBED_VERSION >= 51000)  // 5.10 or above
-            #define PAL_FS_MOUNT_POINT_SECONDARY    "/default2"
-        #else
-            #define PAL_FS_MOUNT_POINT_SECONDARY    "/sd2"
-        #endif
-    #else
-        #define PAL_FS_MOUNT_POINT_SECONDARY PAL_FS_MOUNT_POINT_PRIMARY      //!< User should change this for the his working folder
-    #endif
+    #define PAL_FS_MOUNT_POINT_SECONDARY PAL_FS_MOUNT_POINT_PRIMARY
 #endif
 
+
 #ifndef PAL_NUM_OF_THREAD_INSTANCES
     #define PAL_NUM_OF_THREAD_INSTANCES 1
 #endif
 
 #ifndef PAL_MAX_SEMAPHORE_COUNT
-    #define PAL_MAX_SEMAPHORE_COUNT 1024
-#endif
-
-#ifndef PAL_USE_INTERNAL_FLASH
-    #define PAL_USE_INTERNAL_FLASH  1
-#endif
-
-#ifndef PAL_INT_FLASH_NUM_SECTIONS
-    #define PAL_INT_FLASH_NUM_SECTIONS 2
+    #define PAL_MAX_SEMAPHORE_COUNT 	1024
 #endif
 
 #ifndef PAL_USE_HW_ROT
     #define PAL_USE_HW_ROT 0
 #endif
 
-#ifndef PAL_US_HW_RTC
+#ifndef PAL_USE_HW_RTC
     #define PAL_USE_HW_RTC 0
 #endif
 
-// DEVICE_TRNG is a define that mbed-os defines for every board that is configured to have a TRNG
-// mbedtls uses this define when gathering entropy 
-#if defined(DEVICE_TRNG) && !defined(PAL_USE_HW_TRNG)
+#ifndef PAL_USE_HW_TRNG
     #define PAL_USE_HW_TRNG 1
-#elif !defined(DEVICE_TRNG) && !defined(PAL_USE_HW_TRNG)
-    #define PAL_USE_HW_TRNG 0
 #endif
 
-#endif /* PAL_MBEDOS_CONFIGURATION_H_ */
+#ifndef PAL_USE_INTERNAL_FLASH
+    #define PAL_USE_INTERNAL_FLASH 0
+#endif
+
+#ifndef PAL_USE_SECURE_TIME
+    #define PAL_USE_SECURE_TIME 1
+#endif
+
+//issue a warning if PAL_USE_INTERNAL_FLASH!=0 and or define PAL_USE_INTERNAL_FLASH=0
+#if (PAL_USE_INTERNAL_FLASH == 1) //PAL_USE_INTERNAL_FLASH != 0
+    #warning "Internal flash APIs should be disabled with KVStore"
+#endif
+
+#endif /* PAL_MBEDOS_SST_CONFIGURATION_H_ */
diff --git a/mbed-client-pal/Configs/pal_config/sotp_fs.h b/mbed-client-pal/Configs/pal_config/sotp_fs.h
deleted file mode 100755
index d7d70f0..0000000
--- a/mbed-client-pal/Configs/pal_config/sotp_fs.h
+++ /dev/null
@@ -1,30 +0,0 @@
-// ----------------------------------------------------------------------------
-// Copyright 2016-2019 ARM Ltd.
-//
-// SPDX-License-Identifier: Apache-2.0
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-// ----------------------------------------------------------------------------
-
-#ifndef PAL_HEADER_SOTP_FS
-#define PAL_HEADER_SOTP_FS
-
-
-#define PAL_USE_HW_ROT 1
-#define PAL_USE_HW_RTC 0
-#define PAL_USE_HW_TRNG 1
-#define PAL_SIMULATOR_FLASH_OVER_FILE_SYSTEM 1
-#define PAL_USE_SECURE_TIME 1
-
-
-#endif //PAL_HEADER_SOTP_FS
diff --git a/mbed-client-pal/Source/PAL-Impl/Services-API/pal_configuration.h b/mbed-client-pal/Source/PAL-Impl/Services-API/pal_configuration.h
index 033db1b..45d6528 100644
--- a/mbed-client-pal/Source/PAL-Impl/Services-API/pal_configuration.h
+++ b/mbed-client-pal/Source/PAL-Impl/Services-API/pal_configuration.h
@@ -32,8 +32,6 @@
 
 #ifdef PAL_USER_DEFINED_CONFIGURATION
     #include PAL_USER_DEFINED_CONFIGURATION
-#else
-    #include "sotp_fs.h"
 #endif
 
 /*! \file pal_configuration.h
-- 
2.17.1

